pkg/envoy/proxy.go

```diff
@@ -392,7 +353,7 @@ func (p *Proxy) buildCommand(ctx context.Context, cfgPath string) *exec.Cmd {
 
 	args := append(
 		[]string{
-			"--config-path", cfgPath,
+			"--config-yaml", cfgYaml,
```

Comment:
Alongside the `setcap` changes applied in #238 , this is the meat of the change. Instead of writing the config to disk at `os.TempDir()` and then passing the filepath to the `envoy` process, the config yaml -- or JSON in our case -- is passed directly. JSON is allowed here according to the [envoy docs](https://www.envoyproxy.io/docs/envoy/latest/operations/cli#cmdoption-config-yaml).